app/models/user.rb

```diff
@@ -521,6 +521,10 @@ def reload(...)
     super(...)
   end
 
+  def last_sign_in_email_address
+    confirmed_email_addresses.first
+  end
```

Comment:
Similar comment about test coverage in `spec/models/user_spec.rb`. I'd be specifically concerned to have regression coverage in how we rely on `confirmed_email_addresses` to be ordered here, which I wouldn't think is a given.

Reply:
IMO, it would be better to use scopes as shown at the end of https://api.rubyonrails.org/v8.0.1/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope. Should I go ahead?

Reply:
What would scopes do for us here, or at least specifically in regard to my comment?

Reply:
We would be explicit when we say `user.email_addresses.confirmed.last_signed_in`.